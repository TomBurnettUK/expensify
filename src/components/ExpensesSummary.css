.summary-header {
  padding: 1rem;
}

.page-header-title {
  line-height: 2rem;
}

.page-header-title span {
  font-weight: 700;
}

.summary-buttons {
  margin-top: 1.5rem;
}

.add-button {
  background-color: #364051;
  color: #FFF;
  border: none;
  display: inline-block;
  padding: 1rem 1.5rem;
  font-size: 1.1rem;
  font-weight: 300;
  text-decoration: none;
  line-height: 1;
}

@media