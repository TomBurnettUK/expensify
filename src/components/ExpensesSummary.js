import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import 'numeral/locales/en-gb';

import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

import './ExpensesSummary.css';

numeral.locale('en-gb');

export const ExpensesSummary = props => (
  <div className="page-header summary-header">
    <div className="content-container">
      <h2 className="page-header-title">
        Viewing <span>{props.expenseCount}</span> expenses, totalling{' '}
        <span>{numeral(props.expensesTotal / 100).format('$0,0.00')}</span>
      </h2>
      <div className="summary-buttons">
        <Link className="add-button" to="/create">
          Add Expense
        </Link>
      </div>
    </div>
  </div>
);

const mapStateToProps = state => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: getExpensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpensesSummary);
