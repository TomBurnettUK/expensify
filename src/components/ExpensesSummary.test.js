import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from './ExpensesSummary';

it('should render correctly with empty values', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={0} expensesTotal={0} />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should render correctly with populated values', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={3} expensesTotal={1000} />
  );
  expect(wrapper).toMatchSnapshot();
});
