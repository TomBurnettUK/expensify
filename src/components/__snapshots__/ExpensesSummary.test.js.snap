// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with empty values 1`] = `
<div>
  <p>
    Viewing 
    0
     expenses, totalling
     
    £0.00
  </p>
</div>
`;

exports[`should render correctly with populated values 1`] = `
<div>
  <p>
    Viewing 
    3
     expenses, totalling
     
    £10.00
  </p>
</div>
`;
